pkg/controller/clusterdeprovision/clusterdeprovision_controller.go

```diff
@@ -135,6 +137,32 @@ func (r *ReconcileClusterDeprovision) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, nil
 	}
 
+	// Check if there is a ClusterDeployment owning this Deprovision, if so look it up and
+	// make sure it has a deletion timestamp. Otherwise bail out as a safety check.
+	oRef := metav1.GetControllerOf(instance)
+	if oRef == nil {
+		// TODO: this was once supported to cleanup self-managed "preserveOnDelete" clusters,
+		// but the feature was killed off. For now we'd rather not open the door to dangling
+		// ClusterDeprovisions with no associated cluster.
+		rLog.Warn("ClusterDeprovision does not have an owning ClusterDeployment")
+		return reconcile.Result{}, fmt.Errorf("ClusterDeprovision does not have an owning ClusterDeployment")
+	}
+	if oRef.Kind != "ClusterDeployment" || !strings.HasPrefix(oRef.APIVersion, "hive.openshift.io") {
+		rLog.Warnf("ClusterDeprovision has a non-ClusterDeployment owner: %v", oRef)
+		return reconcile.Result{}, fmt.Errorf("ClusterDeprovision has a non-ClusterDeployment owner: %v", oRef)
+	}
+	cd := &hivev1.ClusterDeployment{}
+	err = r.Get(context.TODO(), types.NamespacedName{Namespace: instance.Namespace, Name: oRef.Name}, cd)
+	if err != nil {
+		rLog.Error("error looking up ClusterDeployment that owns ClusterDeprovision")
+		return reconcile.Result{}, nil
+	}
+	if cd.DeletionTimestamp == nil {
+		err = fmt.Errorf("ClusterDeprovision created for ClusterDeployment that has not been deleted")
+		rLog.WithError(err).Error("skipping")
```

Comment:
I don't see any need to use `WithError` here. Make the message of the log entry the text of what you have as the error.